src/Tasks.UnitTests/Hash_Tests.cs

```diff
@@ -43,6 +43,110 @@ public void HashTaskEmptyInputTest()
             Assert.Null(zeroLengthItemsHash);
         }
 
+        [Fact]
+        public void HashTaskIgnoreOrderTest()
+        {
+            var variant1 =
+                ExecuteHashTask(new ITaskItem[]
+                    {
+                        new TaskItem("item1"),
+                        new TaskItem("item2"),
+                        new TaskItem("item3")
+                    },
+                    ignoreOrder: true);
+            var variant2 =
+                ExecuteHashTask(new ITaskItem[]
+                    {
+                        new TaskItem("item2"),
+                        new TaskItem("item1"),
+                        new TaskItem("item3")
+                    },
+                    ignoreOrder: true);
+            var variant3 =
+                ExecuteHashTask(new ITaskItem[]
+                    {
+                        new TaskItem("item1"),
+                        new TaskItem("item3"),
+                        new TaskItem("item2")
+                    },
+                    ignoreOrder: true);
+            Assert.Equal(variant1, variant3);
+            Assert.Equal(variant1, variant2);
+            Assert.Equal(variant2, variant3);
+        }
+
+        [Fact]
+        public void HashTaskIgnoreOrderNegativeTest()
+        {
+            var variant1 =
+                ExecuteHashTask(new ITaskItem[]
+                    {
+                        new TaskItem("item1"),
+                        new TaskItem("item2"),
+                        new TaskItem("item3")
+                    });
+            var variant2 =
+                ExecuteHashTask(new ITaskItem[]
+                    {
+                        new TaskItem("item2"),
+                        new TaskItem("item1"),
+                        new TaskItem("item3")
+                    });
+            Assert.NotEqual(variant1, variant2);
+        }
+
+        [Fact]
+        public void HashTaskIgnoreCaseNegativeTest()
+        {
+            var uppercaseHash =
+                ExecuteHashTask(new ITaskItem[]
+                    {
+                        new TaskItem("ITEM1"),
+                        new TaskItem("ITEM2"),
+                        new TaskItem("ITEM3")
+                    });
+            var mixedcaseHash =
+                ExecuteHashTask(new ITaskItem[]
+                    {
+                        new TaskItem("Item1"),
+                        new TaskItem("iTEm2"),
+                        new TaskItem("iteM3")
+                    });
+            Assert.NotEqual(uppercaseHash, mixedcaseHash);
+        }
+
+        [Fact]
+        public void HashTaskIgnoreOrderIgnoreCaseTest()
+        {
+            var variant1 =
+                ExecuteHashTask(new ITaskItem[]
+                    {
+                        new TaskItem("ITEM1"),
+                        new TaskItem("ITEM2"),
+                        new TaskItem("ITEM3")
+                    },
+                    true, true);
```

Comment:
nit: can we add parameter names like the previous tests?

Reply:
Done.

Reply:
update https://github.com/dotnet/msbuild/pull/6522/files/fef9cb909ac83829aefedd591b84429554168b6f#diff-a6128683a8ea7da2a38611bf6205d3e5bc60bd7e906853a026b628c1f0987af3